core/src/dsnp/dsnp_types.rs

```diff
@@ -0,0 +1,168 @@
+use serde::{
+	de::{SeqAccess, Visitor},
+	Deserialize, Deserializer, Serialize, Serializer,
+};
+use std::{cmp, error::Error, fmt};
+
+/// Prid len in bytes
+const PRID_LEN_IN_BYTES: usize = 8;
+/// Inner Graph type used in both private and public graphs
+pub type DsnpInnerGraph = Vec<DsnpGraphEdge>;
+/// `Pseudonymous Relationship Identifier` which allows private connection verification
+/// Wrapping in its own Struct allows easier serialization and deserialization
+#[derive(Debug, Clone, PartialEq)]
+pub struct DsnpPrid {
+	inner: Vec<u8>,
```

Comment:
Since we have an upper bound of `PRID_LEN_IN_BYTES`, should we use `BoundedVec<u8, PRID_LEN_IN_BYTES>` here?

Reply:
Ah, right, forgot that was not a standard Rust struct.